app/src/main/java/com/codesoom/assignment/TaskController.java

```diff
@@ -0,0 +1,163 @@
+            List<Task> tasks = this.taskService.showAll();
+
+            if (tasks.size() > 0) {
+                sendResponse(exchange, 200, toJSON(tasks));
+            }
+
+            sendResponse(exchange, 404, "There's no registered tasks");
+
+        } else {
+            sendResponse(exchange, 400, "This request can not be properly handled");
+        }
+    }
+
+    private void postMapper(@NotNull HttpExchange exchange) {
+        String reqPath = exchange.getRequestURI().getPath();
+        if (isReqRegisterOneTask(reqPath)) {
+            /**
+             * TODO
+             * 1. exchange에서 req body 받아와서 Task에 mapping
+             * 2. taskService.register 메소드 호출
+             * 3. 결과 response로 만들어서 반환
+             */
+        }
+    }
+
+    private void putMapper(@NotNull HttpExchange exchange) {
+        String reqPath = exchange.getRequestURI().getPath();
+        if (isReqModifyOneTask(reqPath)) {
+            /**
+             * TODO
+             * 1. exchange에서 req body 받아와서 Task에 mapping,
+             * 2. taskService.modify 메소드를 id와 함께 호출
+             * 3. 결과 response로 만들어서 반환
+             */
+        }
+    }
+
+    private void deleteMapper(@NotNull HttpExchange exchange) {
+        /**
+         * TODO
+         * 1. taskService.delete 메소드를 id와 함께 호출
+         * 2. 결과 response로 만들어서 반환
+         */
+    }
+
+    private void sendResponse(HttpExchange exchange, Integer statusCode, String content) {
+        try {
+            // Response Header
+            exchange.sendResponseHeaders(statusCode, content.getBytes().length);
+
+            // Response Body
+            OutputStream outputStream = exchange.getResponseBody();
+            outputStream.write(content.getBytes());
+
+            // Send and Close
+            outputStream.flush();
+            outputStream.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String toJSON(Task task) throws IOException {
+        OutputStream outputStream = new ByteArrayOutputStream();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.writeValue(outputStream, task);
+
+        return outputStream.toString();
+    }
+
+    private String toJSON(List<Task> tasks) throws IOException {
+        OutputStream outputStream = new ByteArrayOutputStream();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.writeValue(outputStream, tasks);
+
+        return outputStream.toString();
+    }
+
+    private Boolean isReqGettingAllTasks(String path) {
+        return path.matches("/tasks/?");
+    }
+
+    private Boolean isReqGettingOneTask(String path) {
+        return path.matches("/tasks/\\d+/?");
+    }
+
+    private Boolean isReqRegisterOneTask(String path) {
+        return path.matches("/tasks/?");
+    }
+
+    private Boolean isReqModifyOneTask(String path) {
+        return path.matches("/tasks/\\d+?");
+    }
```

Comment:
`?`가 없어도 문제가 없을 것 같습니다.
```suggestion
    private Boolean isReqModifyOneTask(String path) {
        return path.matches("/tasks/\\d+");
    }
```

Reply:
`/tasks/\\d+/?`라고 적는다는걸 슬래쉬를 하나 빠뜨렸군요 😅 request path의 마지막에 슬래쉬가 포함되는 케이스가 있을 것 같아서 `/?`를 추가했었는데, 이 부분은 고려하지 않아도 괜찮을까요?

Reply:
이건 이 링크로 제 입장을 대변하도록 하죠. https://johngrib.github.io/wiki/Postel-s-law/

Reply:
가능한 관대하게 받기 위해 마지막의 `/` 여부까지 고려하는 것으로 처리하는 것이 좋겠군요 ㅎㅎ 👍